### PR TITLE
Automatically add test names

### LINKED ISSUE #8: Support automatic test case naming
This deserves a bit of thought before going forward

*  right now the fallback is to have no name, which obviously isn't ideal
*  deparsing might be the right next step
*  i also need something like snapshot testing to actually look at the testcase output to be sure that it's sensible

### PR DESCRIPTION
**Pull request by michaelquinn32**
Fixes #8 and #9

### CODE REVIEW

**Comment by michaelquinn32** [comment]
@MichaelChirico How does this look? Thanks!

File: R/with_parameters.R [hunk 1]
```diff
@@ -27,9 +27,10 @@
 #' data frame, pass it in the `.cases` argument.
 #'
 #' One parameter is noteworthy. If the user passes a character vector as
-#' `test_name`, each instance is combined with `desc_stub` to create the
+#' `.test_name`, each instance is combined with `desc_stub` to create the
 #' completed test name. Similarly, the named argument from `cases()` is combined
-#' with `desc_stub` to create the parameterized test names.
+#' with `desc_stub` to create the parameterized test names. When names aren't
+#' provided, they will be automatically generated using the test data.
```
Comment (MichaelChirico) [inline R/with_parameters.R]: would help to mention how it's constructed so user can more quickly decide if they're OK w auto-naming

File: R/with_parameters.R [hunk 2]
```diff
@@ -109,11 +111,30 @@ with_parameters_test_that <- function(desc_stub,
...
+#' @param all_cases A tibble containing test cases.
+#' @return A character vector, whose length matches the number of rows in
+#'   `all_cases`.
+#' @noRd
+build_test_names <- function(all_cases) {
+  case_names <- names(all_cases)
+  build_label <- function(...) {
+    row <- format(list(...))
+    labels <- sprintf("%s=%s", case_names, row)
+    paste(labels, collapse = ", ")
```
Comment (MichaelChirico) [inline R/with_parameters.R]: toString() 🙃

**Review by MichaelChirico — COMMENTED** [review]
minor comments only, basic idea is great!

File: R/with_parameters.R (hunk 1 above)
Comment (michaelquinn32) [reply R/with_parameters.R]: Great point.

File: R/with_parameters.R (hunk 2 above)
Comment (michaelquinn32) [reply R/with_parameters.R]: Bah

Comment (MichaelChirico) [reply R/with_parameters.R]: do you want me to add a linting GHA?

**Comment by MichaelChirico** [comment]
WDYT about an extension here where `.test_names=` can be a glue-formatted string?

```r
with_parameters_test_that("str", { ... },
  .test_names = "{x} case where {y}, {z}",
  x = ...
  y = ...
  z = ...
)
```

The actual test name is filled out with `x[i], y[i], z[i]`.

Not fully back-compatible in the case that (1) `.test_names` were previously supplied with `{` _and_ (2) the symbol inside `{...}` resolves to a variable in the parameters. But I don't think it'll break anything, just weird test names maybe, and seems rare anyway.

**Pull request merged**